app/models/unit_template.rb

```diff
@@ -19,6 +19,10 @@ def activity_ids= activity_ids
     self.activities = Activity.find(activity_ids)
   end
 
+  def self.related_models(user, unit_template)
```

Comment:
@gauravrakheja Since this method requires an instance of a unit_template why don't we make it an instance method instead of a class method? That way we only need to pass the user.

We could further simplify by only passing the flag, Easier to test and a bit more abstract.